tests/test-core/testTests/js/jqUnitTests.js

```diff
@@ -0,0 +1,70 @@
+/*
+ Copyright 2016 Raising the Floor - International
+
+ Licensed under the Educational Community License (ECL), Version 2.0 or the New
+ BSD license. You may not use this file except in compliance with one these
+ Licenses.
+ You may obtain a copy of the ECL 2.0 License and BSD License at
+ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+ */
+
+/* global fluid, jqUnit, QUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests.jqUnit");
+    
+    jqUnit.module("Conventional jqUnit tests");
+    
+    fluid.tests.jqUnit.failIsPass = false;
+    
+    // We have to use this insane scheme for subverting QUnit's accounting rather than a teardown function,
+    // since we need to beat qunit-composite.js own logging function. When will the "industry" learn that
+    // integration is the first priority?
+    QUnit.log(function (data) {
```

Comment:
It's because qunit-composite.js uses a completely different strategy to the standard QUnit UI for accounting for test failures: https://github.com/fluid-project/infusion/blob/master/tests/lib/qunit/addons/composite/qunit-composite.js#L79 . If we just use failingTeardown, the main QUnit UI works fine, but the "failures" are rendered as real failures in the iframe-based all-tests.html driver.